### PR TITLE
Kernel: Add support for QEMU ISA-PC & QEMU MicroVM machine types

### PR DESCRIPTION
**Pull request by supercomputer7**
This PR sets a goal to create basic abstractions in the kernel to allow us to boot SerenityOS on a pure ISA-PC machine type.

### CODE REVIEW

**Comment by supercomputer7** [comment]
I know this could be a questionable patch (because SerenityOS is not about supporting really *really* old stuff like ISA-only PCs), so I'll quote one of the commit messages here:

> This is mainly a good environment for testing abstractions in the kernel
space, and can help with improving on them for the sake of porting the
OS to other chipsets and CPU architectures.

Another goal I have in the future is to support the QEMU `microvm` machine type, which doesn't support PCI nor ACPI by default. That type of machine could be used on CI machines to test userland utilities without holding us to have the overhead of a fully-emulated chipset like the i440FX or Q35 machines. It really can help also to have good abstractions for platforms that don't use PCI at all (so everything is memory mapped and there's a device tree, like for the `microvm` machine).
This might also assit some people (like me) to test the system on bare metal, if we have a bug/problem in the PCI subsystem/PCI based driver in the kernel.

File: Kernel/Storage/ATA/ISAIDEController.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2020, Liav A. <liavalb@hotmail.co.il>
```
Comment (supercomputer7) [inline Kernel/Storage/ATA/ISAIDEController.h:2]: Probably should update the year here...

File: Kernel/Storage/ATA/PCIIDEController.cpp [hunk 2]
```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2020, Liav A. <liavalb@hotmail.co.il>
```
Comment (supercomputer7) [inline Kernel/Storage/ATA/PCIIDEController.cpp:2]: Probably should update the year here...

File: Kernel/Storage/ATA/PCIIDEController.h [hunk 3]
```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2020, Liav A. <liavalb@hotmail.co.il>
```
Comment (supercomputer7) [inline Kernel/Storage/ATA/PCIIDEController.h:2]: Probably should update the year here...

File: Kernel/Graphics/VGA/ISAAdapter.cpp [hunk 4]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2021, Liav A. <liavalb@hotmail.co.il>
```
Comment (supercomputer7) [inline Kernel/Graphics/VGA/ISAAdapter.cpp:2]: Probably should update the year here...

File: Kernel/Graphics/VGA/ISAAdapter.h [hunk 5]
```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2021, Liav A. <liavalb@hotmail.co.il>
```
Comment (supercomputer7) [inline Kernel/Graphics/VGA/ISAAdapter.h:2]: Probably should update the year here...

File: Kernel/Graphics/VGA/PCIAdapter.h [hunk 6]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2021, Liav A. <liavalb@hotmail.co.il>
```
Comment (supercomputer7) [inline Kernel/Graphics/VGA/PCIAdapter.h:2]: Probably should update the year here...

**Comment by supercomputer7** [comment]
A "wild" idea that was suggested by @boricj is to write support for FDT (Flattened Device Tree) support and use it together with the ISA-PC machine type (this could be supplied as boot module from the multiboot bootloader) to know which devices exist in such environment. This in theory provide great platform to expand for other architectures which rely on FDTs instead of ACPI.
Keeping this note here as we might want to implement it sometime in the future :)

**Comment by supercomputer7** [comment]
Making this a draft as I tested this with #11368 and a couple of problems were discovered when I tried it together.

**Comment by awesomekling** [comment]
> This branch has conflicts that must be resolved

**Comment by supercomputer7** [comment]
> > This branch has conflicts that must be resolved

Rebased to solve the merge conflict.

**Pull request merged**